src/vm/gcenv.ee.cpp

```diff
@@ -868,7 +870,7 @@ void GCToEEInterface::StompWriteBarrier(WriteBarrierParameters* args)
         }
 #endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
 
-        ::StompWriteBarrierResize(args->is_runtime_suspended, args->requires_upper_bounds_check);
+        flushXrestart |= ::StompWriteBarrierResize(args->is_runtime_suspended, args->requires_upper_bounds_check);
```

Comment:
This feels like it's potentially trouble and needs close scrutiny. A checked write barrier (like [this one](https://github.com/dotnet/coreclr/blob/master/src/vm/amd64/JitHelpers_Fast.asm#L436)) will observe the below writes to `g_lowest_address` and `g_highest_address`, but it will not observe the write to the instruction-patched `g_card_table`, because we are no longer flushing the icache. The card table is "translated" in such a way that indexing into it is relative to `g_lowest_address` (it's constructed such that `g_card_table[g_lowest_address]` is the zeroth entry in the card table) so reading a stale `g_card_table` is sure to be trouble.

Reply:
Shame on me I didn't pay attention to that detailed comment below. Now fixed. Also patched ARM64 code below which I missed altogether. Fortunate consequense of returning flags from these methods is that we can skip such expensive calls if card table location wasn't patched.